src/main/java/com/mixpanel/android/viewcrawler/ImageStore.java

```diff
@@ -0,0 +1,138 @@
+package com.mixpanel.android.viewcrawler;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.util.Base64;
+import android.util.Log;
+
+import com.mixpanel.android.util.HttpService;
+import com.mixpanel.android.util.RemoteService;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * ABSOLUTELY NOT thread, or even process safe.
+ * Writes and reads files and directories at known paths, and uses a shared instance of MessageDigest.
+ */
+/* package */ class ImageStore {
+    public ImageStore(Context context) {
+        this(
+            context.getDir(DEFAULT_DIRECTORY_NAME, Context.MODE_PRIVATE),
+            new HttpService()
+        );
+    }
+
+    public ImageStore(File directory, RemoteService poster) {
+        mDirectory = directory;
+        mPoster = poster;
+        MessageDigest useDigest;
+        try {
+            useDigest = MessageDigest.getInstance("SHA1");
+        } catch (NoSuchAlgorithmException e) {
+            Log.w(LOGTAG, "Images won't be stored because this platform doesn't supply a SHA1 hash function");
+            useDigest = null;
+        }
+
+        mDigest = useDigest;
+    }
+
+    public Bitmap getImage(String url) {
```

Comment:
This should throw on failure, not return null, for consistency with EditProtocol.